lib/tools/apk-signing.js

```diff
@@ -103,22 +135,33 @@ apkSigningMethods.zipAlignApk = async function (apk) {
  * @return {boolean} True if given application is already signed.
  */
 apkSigningMethods.checkApkCert = async function (apk, pkg) {
-  const java = getJavaForOs();
+  log.debug(`Checking app cert for ${apk}`);
   if (!(await fs.exists(apk))) {
     log.debug(`APK doesn't exist. ${apk}`);
     return false;
   }
   if (this.useKeystore) {
     return await this.checkCustomApkCert(apk, pkg);
   }
-  log.debug(`Checking app cert for ${apk}.`);
+
+  let verificationFunc;
+  try {
+    const apkSigner = await this.getBinaryFromSdkRoot(`apksigner${system.isWindows() ? '.bat' : ''}`);
+    const cmd = [apkSigner, 'verify', apk];
+    verificationFunc = async () => await exec(system.isWindows() ? 'cmd.exe' : 'bash', cmd);
+  } catch (e) {
+    log.warn(`Cannot use apksigner tool for signature verification. Defaulting to verify.jar. ` +
+      `Original error: ${e.message}`);
+    const java = getJavaForOs();
+    verificationFunc = async () => await exec(java, ['-jar', path.resolve(this.helperJarPath, 'verify.jar'), apk]);
+  }
   try {
-    await exec(java, ['-jar', path.resolve(this.helperJarPath, 'verify.jar'), apk]);
-    log.debug("App already signed.");
+    await verificationFunc();
     await this.zipAlignApk(apk);
```

Comment:
@imurchie I'm not sure about whether we should execute zipAlignApk here. To me it looks more like an unexpected side effect in scope of this method

Reply:
I'd think we always want to zip align the apk?

Reply:
but aligning the apk breaks the signature, which means this method is going to brake the signature after it has been successfully verified.

Reply:
It sounds like there's no issue if zipalign happens before signing.

Reply:
so you don't mind if I remove this call from `checkApkCert` method?